MultimodalQnA/ui/gradio/multimodalqna_ui_gradio.py

```diff
@@ -315,6 +315,64 @@ def ingest_image_gen_caption(filepath, request: gr.Request):
     return
 
 
+def ingest_image_with_text(filepath, text, request: gr.Request):
+    yield (gr.Textbox(visible=True, value="Please wait for your uploaded image to be ingested into the database..."))
+    verified_filepath = os.path.normpath(filepath)
+    if not verified_filepath.startswith(tmp_upload_folder):
+        print("Found malicious image file name!")
+        yield (
+            gr.Textbox(
+                visible=True,
+                value="Your uploaded image's file name has special characters that are not allowed. Please consider updating the file name!",
+            )
+        )
+        return
+    basename = os.path.basename(verified_filepath)
+    dest = os.path.join(static_dir, basename)
+    shutil.copy(verified_filepath, dest)
+    text_basename = "{}.txt".format(os.path.splitext(basename)[0])
+    text_dest = os.path.join(static_dir, text_basename)
+    with open(text_dest, "a") as file:
+        file.write(text)
+    print("Done copying uploaded files to static folder!")
+    headers = {
+        # 'Content-Type': 'multipart/form-data'
+    }
+    files = [
+        ('files', (basename, open(dest, "rb"))),
+        ('files', (text_basename, open(text_dest, "rb")))
+    ]
+    response = requests.post(dataprep_ingest_addr, headers=headers, files=files)
+    print(response.status_code)
+    os.remove(text_dest)
+    if response.status_code == 200:
+        response = response.json()
+        print(response)
+        yield (gr.Textbox(visible=True, value="Image ingestion is done. Saving your uploaded image..."))
+        time.sleep(2)
```

Comment:
What's the sleep for? Just wondering if timing could vary by machine and if there's a different way to check for something to be ready.

Reply:
I don't know, it's just what they did before 😄 . It's probably for aesthetic reasons. I could try suppressing it and see what it looks like.

Reply:
I think this 2 second pause is for the user to have a chance to read the message that ingestion was successful by the megaservice/backend. The next operation (copying the file to the UI container with the updated unique identifier) will take a very short time, so if there were no `time.sleep(2)` the next message "Congratulation! Your upload is done!..." would overwrite the other message almost instantly.